### PR TITLE
Added primary output option in DocumentNodeType

### LINKED ISSUE #1235: Allow nodes to have no primary output
The primary input is optional for nodes, but the primary output is currently required. The Split Channels node needs to have no primary output. Add support for this, and remove that node's primary output connector.

### PR DESCRIPTION
**Pull request by gautamprikshit1**
Closes #1235

### CODE REVIEW

**Comment by Keavon** [comment]
Could you please rearrange the `primary_output` field to appear on the line below the `outputs` field?

**Comment by gautamprikshit1** [comment]
Including type definition?

File: editor/src/messages/portfolio/document/node_graph/node_graph_message_handler/document_node_types.rs [hunk 1]
```diff
@@ -86,6 +86,7 @@ pub struct DocumentNodeType {
 	pub inputs: Vec<DocumentInputType>,
 	pub outputs: Vec<DocumentOutputType>,
 	pub properties: fn(&DocumentNode, NodeId, &mut NodePropertiesContext) -> Vec<LayoutGroup>,
+	pub primary_output: bool,
```
Comment (Keavon) [inline editor/src/messages/portfolio/document/node_graph/node_graph_message_handler/document_node_types.rs:89]: Here too please.

**Comment by Keavon** [comment]
You also have merge conflicts, could you please rebase your branch onto the latest master?

**Comment by TrueDoctor** [comment]
> Could you please rearrange the `primary_output` field to appear on the line below the `outputs` field on every changed line?

We should also think about  implementing Default for DocumentNodeType and use the `..Default::default()` instead to reduce bloat

**Comment by Keavon** [comment]
Yeah, that would be a good change.

**Review by TrueDoctor — APPROVED** [review]

**Pull request merged**